CoreGraphFramework/test/unit/src/au/gov/asd/tac/constellation/graph/attribute/BooleanAttributeDescriptionNGTest.java

```diff
@@ -0,0 +1,461 @@
+import org.testng.annotations.Test;
+
+/**
+ *
+ * @author arcturus
+ */
+public class BooleanAttributeDescriptionNGTest {
+
+    BooleanAttributeDescription instance;
+
+    public BooleanAttributeDescriptionNGTest() {
+    }
+
+    @BeforeClass
+    public static void setUpClass() throws Exception {
+    }
+
+    @AfterClass
+    public static void tearDownClass() throws Exception {
+    }
+
+    @BeforeMethod
+    public void setUpMethod() throws Exception {
+        instance = new BooleanAttributeDescription();
+        instance.setCapacity(1);
+    }
+
+    @AfterMethod
+    public void tearDownMethod() throws Exception {
+    }
+
+    /**
+     * Test of getName method, of class BooleanAttributeDescription.
+     */
+    @Test
+    public void testGetName() {
+        String expResult = "boolean";
+        String result = instance.getName();
+        assertEquals(result, expResult);
+    }
+
+    /**
+     * Test of getNativeClass method, of class BooleanAttributeDescription.
+     */
+    @Test
+    public void testGetNativeClass() {
+        Class expResult = boolean.class;
+        Class result = instance.getNativeClass();
+        assertEquals(result, expResult);
+    }
+
+    /**
+     * Test of getNativeType method, of class BooleanAttributeDescription.
+     */
+    @Test
+    public void testGetNativeType() {
+        NativeAttributeType expResult = NativeAttributeType.BOOLEAN;
+        NativeAttributeType result = instance.getNativeType();
+        assertEquals(result, expResult);
+    }
+
+    /**
+     * Test of getDefault method, of class BooleanAttributeDescription.
+     */
+    @Test
+    public void testGetDefault() {
+        Object expResult = false;
+        Object result = instance.getDefault();
+        assertEquals(result, expResult);
+    }
+
+    /**
+     * Test of setDefault method, of class BooleanAttributeDescription.
+     */
+    @Test
+    public void testSetDefault() {
+        Object value = null;
+        instance.setDefault(value);
+    }
+
+    /**
+     * Test of getCapacity method, of class BooleanAttributeDescription.
+     */
+    @Test
+    public void testGetCapacity() {
+        int expResult = 1;
+        int result = instance.getCapacity();
+        assertEquals(result, expResult);
+    }
+
+    /**
+     * Test of setCapacity method, of class BooleanAttributeDescription.
+     */
+    @Test
+    public void testSetCapacity() {
+        int capacity = 0;
+        instance.setCapacity(capacity);
+    }
+
+    /**
+     * Test of getByte method, of class BooleanAttributeDescription.
+     */
+    @Test
+    public void testGetByte() {
+        int id = 0;
+        byte expResult = 0;
+        byte result = instance.getByte(id);
+        assertEquals(result, expResult);
+    }
+
+    /**
+     * Test of setByte method, of class BooleanAttributeDescription.
+     */
+    @Test
+    public void testSetByte() {
+        int id = 0;
+        byte value = 0;
+        instance.setByte(id, value);
```

Comment:
Should this have an assert to check if it was set correctly?
Quite a few more it seems, I wont add review for all. If it does need attention then I'd say just go through the tests.

Reply:
Yep updated most of them and ready for a review @aldebaran30701. I might continue with more tests down the track but would like to get whats done in if thats ok.